backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/impl/LuceneSearcherImpl.java

```diff
@@ -165,7 +165,7 @@ private LuceneCollectors buildCollectors(IndexSearcher indexSearcher, IndexReade
 	}
 
 	private int getMaxDocs(IndexReader reader, int offset, Integer limit) {
-		if ( limit == null ) {
+		if ( limit == null || (long) offset + limit >= Integer.MAX_VALUE ) {
```

Comment:
Why not just `limit.equals( Integer.MAX_VALUE )`? Do you also want to ignore the limit when using `offset = 5` and `limit = Integer.MAX_VALUE - 5`?

If you want to take the offset into account, it would be better to take it into account in the Elasticsearch backend as well.